### PR TITLE
feat(builder): add max min length in string option

### PR DESCRIPTION
**Pull request by imranbarbhuiya**
**Please describe the changes this PR makes and why it should be merged:**

- discord/discord-api-docs#5150

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/builders/package.json
Comment (vladfrangu) [inline packages/builders/package.json:56]: If we bump it here without bumping it everywhere we'll have version clashes again :pepehands

File: packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandStringOptionMinMaxLengthMixin.ts
Comment (vladfrangu) [inline packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandStringOptionMinMaxLengthMixin.ts:1]: Can't you just re-use the number/integer one but renamed so its more common?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandStringOptionMinMaxLengthMixin.ts
Comment (imranbarbhuiya) [reply packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandStringOptionMinMaxLengthMixin.ts:1]: numeric one is for value and this one is for length.

Comment (vladfrangu) [reply packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandStringOptionMinMaxLengthMixin.ts:1]: Oh woops, didn't read the property right

**Comment by almeidx** [comment]
Can you update the max length?
- discord/discord-api-docs#5150

File: packages/builders/__tests__/interactions/SlashCommands/Options.test.ts
Comment (vladfrangu) [inline packages/builders/__tests__/interactions/SlashCommands/Options.test.ts:175]: Already done, todo is no longer needed

File: yarn.lock
Comment (vladfrangu) [inline yarn.lock:6809]: What's...still using 0.33.3 in the project? Can you run yarn why on it? 😓

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by Jiralite — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (imranbarbhuiya) [reply yarn.lock:6817]: image
should I nuke yarn.lock and regen it?

Comment (vladfrangu) [reply yarn.lock:6817]: Why is rest coming from npm...

**Review by vladfrangu — APPROVED** [review]

File: yarn.lock
Comment (imranbarbhuiya) [reply yarn.lock:6817]: bcz /proxy don't uses yarn workspace for publish

File: packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandStringOptionMinMaxLengthMixin.ts
Comment (suneettipirneni) [inline packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandStringOptionMinMaxLengthMixin.ts:1]: Can't this just be part of the `SlashCommandStringOption`, rather than a mixin since it's only applicable to string options?

**Review by suneettipirneni — CHANGES_REQUESTED** [review]

**Comment by iCrawl** [comment]
So, this might be a bit controversial here but, if you update `-types` in one place, we should probably update it everywhere, especially as long as v14 is not released yet.

**Comment by imranbarbhuiya** [comment]
> So, this might be a bit controversial here but, if you update `-types` in one place, we should probably update it everywhere, especially as long as v14 is not released yet.

I did. Didn't I? Also for /rest https://github.com/discordjs/discord.js/pull/8214#discussion_r913138935

**Comment by iCrawl** [comment]
Guess I am just surprised by the lack of breaking changes 😆 
Thanks for double checking.

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]
LGTM, https://github.com/discordjs/discord.js/labels/blocked until upstream is merged.

**Comment by SuperchupuDev** [comment]
upstream pr was merged 🚀

**Pull request merged**
